packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -415,11 +416,19 @@ export function createElement(
         type,
       );
     }
-
     if (type === 'script') {
       // Create the script via .innerHTML so its "parser-inserted" flag is
       // set to true and it does not execute
       const div = ownerDocument.createElement('div');
+      if (__DEV__) {
+        warningWithoutStack(
+          false,
+          'Encountered script tag while rendering React component. ' +
+            'Scripts inside React components are parser inserted into document ' +
+            'and they are never executed. Furthemore rendering script nodes ' +
+            'inside components breaks when using Trusted Types.',
+        );
+      }
```

Comment:
Is the “parser mode” protection relevant for scripts that have `src` or non standard value of `type` attribute? If not, maybe we can keep allowing those without warning (and not use innerHTML for them)?

Reply:
If you mean Trusted Types, we require typed value for the content regardless of the `type` attribute (or the presence of other attributes), as we'd be running into difficulties (e.g. what happens if the `src` gets cleared after the text has been modified). Discussion under https://github.com/WICG/trusted-types/issues/6.  

If you mean in HTML, it seems like the spec mandates setting the parser inserted flag unconditionally (https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inhead). That said, it seems like Chrome doesn't comply: 

```
data:text/html,<script type=foo>alert(1)</script><script>s=document.scripts[0];document.head.removeChild(s);s.type='';document.head.appendChild(s);</script>
```

It is respected in FF though.